demos/object_detection_demo/python/object_detection_demo.py

```diff
@@ -69,6 +69,12 @@ def build_argparser():
                                    help='Optional. The first image size used for CTPN model reshaping. '
                                         'Default: 600 600. Note that submitted images should have the same resolution, '
                                         'otherwise predictions might be incorrect.')
+    common_model_args.add_argument('--anchors', default=None, type=float, nargs='+',
+                                   help='Optional. A space separated list of anchors. '
+                                        'By default used default anchors for model. Only for YOLOV4 architecture type.')
+    common_model_args.add_argument('--masks', default=None, type=int, nargs='+',
+                                   help='Optional. A space separated list of mask for anchors. '
+                                        'By default used default masks for model. Only for YOLOV4 architecture type.')
```

Comment:
You should report an error if they are set for anything else.

Reply:
Good point. But it better be a warning, not error.

Reply:
Why? warnings are easy to ignore. Errors that are easy to fix should crash demos to make run it correctly

Reply:
If any of these option set for another architecture, it should cause no effect at all. Because demo in that case works correctly, I don't see any reason to abort demo.

Reply:
That would be strange if there was an option that causes no effect

Reply:
```
Only for YOLOV4 architecture type
```

Reply:
Exactly. You claim it is only for YOLOV4, but accept it for any.

Reply:
IMHO, error states that demo cannot be executed with such options set (i.e. incorrect model input shape - demo would just crash if we continue). Warnings - if demo wolud continue to work correctly, but with some assumptions - i.e. set default value , ignore something. 
For this particular case I'd leave warning: "Warning: --mask parameter is ignored as model type is different from YOLOv4"